activerecord/lib/active_record.rb

```diff
@@ -26,6 +26,7 @@
 require "active_support"
 require "active_support/rails"
 require "active_model"
+require "active_job"
```

Comment:
Can we conditionally include this? https://www.rubydoc.info/github/rubygems/rubygems/Gem%2FSpecification.find_by_name

```suggestion
begin
    Gem::Specification.find_by_name "active_job"
    require "active_job"
    rescue Gem::LoadError
       # maybe log out a warning?
    end
```

Then in future calls we can say
```ruby
do_thing if defined?(ActiveJob)
```

Just a thought...

Reply:
We can but this is not we decouple things in Rails. The way I explained it in https://github.com/rails/rails/pull/34901#discussion_r24647389 is the way we do.